test/vm/unit/runtime/storage_tests.cpp

```diff
@@ -49,352 +49,150 @@ TEST_F(RuntimeTest, TransientStorage)
-
-    // nonempty -> empty
-    ctx_.gas_remaining = 2301;
-    store(key, 0);
-    ASSERT_EQ(ctx_.gas_remaining, 2301);
-    ASSERT_EQ(ctx_.gas_refund, 15000);
-    ASSERT_EQ(load(key), 0);
-}
-
-TEST_F(RuntimeTest, StorageBerlinLoadCold)
-{
-    using traits = EvmTraits<EVMC_BERLIN>;
-    auto load = wrap(sload<traits>);
-
-    ctx_.gas_remaining = 2000;
-    ASSERT_EQ(load(key), 0);
-    ASSERT_EQ(ctx_.gas_remaining, 0);
-    ASSERT_EQ(load(key), 0);
-    ASSERT_EQ(ctx_.gas_remaining, 0);
-}
-
-TEST_F(RuntimeTest, StorageBerlinLoadWarm)
-{
-    using traits = EvmTraits<EVMC_BERLIN>;
-    auto load = wrap(sload<traits>);
-
-    host_.access_storage(ctx_.env.recipient, bytes32_from_uint256(key));
-
-    ctx_.gas_remaining = 0;
-    ASSERT_EQ(load(key), 0);
-    ASSERT_EQ(ctx_.gas_remaining, 0);
-}
-
-TEST_F(RuntimeTest, StorageBerlinOriginalEmpty)
-{
-    using traits = EvmTraits<EVMC_BERLIN>;
-    auto load = wrap(sload<traits>);
-    auto store = wrap(sstore<traits>);
-
-    // empty -> nonempty (cold)
-    ctx_.gas_remaining = 22000;
-    store(key, val);
-    ASSERT_EQ(ctx_.gas_remaining, 0);
-    ASSERT_EQ(load(key), val);
-
-    // nonempty -> nonempty (warm)
-    ctx_.gas_remaining = 2301;
-    store(key, val_2);
-    ASSERT_EQ(ctx_.gas_remaining, 2301);
-    ASSERT_EQ(load(key), val_2);
-
-    // nonempty -> empty (warm)
-    ctx_.gas_remaining = 2301;
-    store(key, 0);
-    ASSERT_EQ(ctx_.gas_remaining, 2301);
-    ASSERT_EQ(ctx_.gas_refund, 19900);
-    ASSERT_EQ(load(key), 0);
-}
-
-TEST_F(RuntimeTest, StorageBerlinOriginalNonEmpty)
-{
-    using traits = EvmTraits<EVMC_BERLIN>;
-    auto load = wrap(sload<traits>);
-    auto store = wrap(sstore<traits>);
+TYPED_TEST(RuntimeTraitsTest, StorageLoadCold)
+{
+    using traits = TestFixture::Trait;
+    auto load = TestFixture::wrap(sload<traits>);
+
+    this->ctx_.gas_remaining = [] {
+        if constexpr (is_monad_trait_v<traits>) {
+            if constexpr (traits::monad_rev() >= MONAD_SEVEN) {
+                return 8000;
+            }
+        }
+        if constexpr (traits::evm_rev() <= EVMC_ISTANBUL) {
+            return 0;
+        }
+        else {
+            return 2000;
+        }
+    }();
+    ASSERT_EQ(load(key), 0);
+    ASSERT_EQ(this->ctx_.gas_remaining, 0);
+    ASSERT_EQ(load(key), 0);
+    ASSERT_EQ(this->ctx_.gas_remaining, 0);
+}
+
+TYPED_TEST(RuntimeTraitsTest, StorageLoadWarm)
+{
+    using traits = TestFixture::Trait;
+    auto load = TestFixture::wrap(sload<traits>);
+
+    this->host_.access_storage(
+        this->ctx_.env.recipient, bytes32_from_uint256(key));
+
+    this->ctx_.gas_remaining = 0;
+    ASSERT_EQ(load(key), 0);
+    ASSERT_EQ(this->ctx_.gas_remaining, 0);
+}
+
+TYPED_TEST(RuntimeTraitsTest, StorageOriginalEmpty)
+{
+    using traits = TestFixture::Trait;
+    auto load = TestFixture::wrap(sload<traits>);
+    auto store = TestFixture::wrap(sstore<traits>);
+
+    auto do_test = [&load, &store, &ctx_ = this->ctx_](
```

Comment:
I suppose the `store` and `load` status codes are supposed to be success in this function